.changeset/cyan-tables-poke.md

```diff
@@ -0,0 +1,40 @@
+---
+'astro': minor
+---
+
+Adds experimental SVGO optimization support for SVG assets
+
+Astro now supports automatic SVG optimization using SVGO during build time. This experimental feature helps reduce SVG file sizes while maintaining visual quality, improving your site's performance.
+
+To enable SVG optimization with default settings, add the following to your `astro.config.mjs`:
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    svgo: true,
+  },
+});
+```
+
+To customize optimization, pass a [SVGO configuration object](https://svgo.dev/docs/plugins/):
+
+```js
+export default defineConfig({
+  experimental: {
+    svgo: {
+      plugins: [
+        'preset-default',
+        {
+          name: 'removeViewBox',
+          active: false
+        }
+      ],
+    },
+  },
+});
+```
+
+The feature includes graceful error handling. If optimization fails for any SVG, Astro will fall back to using the original file content.
```

Comment:
@florian-lefebvre This is what I mean re: Case 2. What does this mean wrt getting errors optimizing?

(And then, similarly in the Docs PR here https://github.com/withastro/docs/pull/12637/files#r2489752607 )

Reply:
This was the original behavior but this got changed after my feedback, I believe.